CRM/Event/BAO/Event.php

```diff
@@ -2042,13 +2052,18 @@ public static function checkPermission($eventId = NULL, $type = CRM_Core_Permiss
     static $permissions = NULL;
 
     if (empty($permissions)) {
-      $allEvents = CRM_Event_PseudoConstant::event(NULL, TRUE);
-      $createdEvents = array();
-
-      $session = CRM_Core_Session::singleton();
-      if ($userID = $session->get('userID')) {
-        $createdEvents = array_keys(CRM_Event_PseudoConstant::event(NULL, TRUE, "created_id={$userID}"));
-      }
+      $result = civicrm_api3('Event', 'get', array(
```

Comment:
Wouldn't keeping $allEvents = CRM_Event_PseudoConstant::event(NULL, TRUE); be more efficient than an API call?